File: Crio/crio_programming_interview_problems-master/RankFromStream/RankFromStream.java
import java.util.*;

public class RankFromStream {
    public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    int q = scanner.nextInt();
    Solution solution = new  Solution();

    for(int i = 0 ; i < q ; i++) {
        char ch = scanner.next().charAt(0);
        int x = scanner.nextInt();
        if(ch == '+') {
            solution.track(x);
        } else {
            System.out.println(solution.getRankOfNumber(x));
        }
    }
    }
}

File: Crio/crio_programming_interview_problems-master/RankFromStream/RankFromStream.py
import sys
from Solution import Solution

q = int(sys.stdin.readline().strip())

solution = Solution()
for i in range(q):
    t, x = sys.stdin.readline().strip().split()
    x = int(x)
    if t == '+':
        solution.track(x)
    else:
        sys.stdout.write(str(solution.getRankOfNumber(x)) + '\n')

